src/pocketmine/Player.php

```diff
@@ -1184,6 +1184,7 @@ public function setSpawn(Vector3 $pos){
 		$pk->y = (int) $this->spawnPosition->y;
 		$pk->z = (int) $this->spawnPosition->z;
 		$pk->spawnType = SetSpawnPositionPacket::TYPE_PLAYER_SPAWN;
+		$pk->spawnForced = true;
```

Comment:
I'm not sure this should be true... spawnForced means players always spawn at their spawn point when they join the server.

Other than that, looks good.